compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -271,6 +271,18 @@ object Contexts {
       if (this.settings.YtraceContextCreation.value)
         creationTrace = (new Throwable).getStackTrace().take(20)
 
+    /** Whether current prettyprint is for -Xprint
+     *  -Xprint will print `sym.name` instead of `sym.originalName`
+     */
+    private var _isXprintMode = false
```

Comment:
Thanks a lot for the explanation and tip, it makes perfect sense. I've now updated the PR based on the `property`-based approach.